src/AWS.Deploy.Common/DeploymentManifest/DeploymentManifestEngine.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using AWS.Deploy.Common.IO;
+using Newtonsoft.Json;
+
+namespace AWS.Deploy.Common.DeploymentManifest
+{
+    public interface IDeploymentManifestEngine
+    {
+        string? TargetApplicationFullPath { get; set; }
+
+        Task UpdateDeploymentManifestFile(string saveCdkDirectoryFullPath);
+    }
+
+    /// <summary>
+    /// This class contains the helper methods to update the deployment manifest file
+    /// that keeps track of the save CDK deployment projects.
+    /// </summary>
+    public class DeploymentManifestEngine : IDeploymentManifestEngine
+    {
+        private readonly IDirectoryManager _directoryManager;
+        private readonly IFileManager _fileManager;
+
+        public string? TargetApplicationFullPath { get; set; }
+
+
+        public DeploymentManifestEngine(IDirectoryManager directoryManager, IFileManager fileManager)
+        {
+            _directoryManager = directoryManager;
+            _fileManager = fileManager;
+        }
+
+        /// <summary>
+        /// This method updates the deployment manifest json file by adding the directory path at which the CDK deployment project is saved.
+        /// If the manifest file does not exists then a new file is generated.
+        /// </summary>
+        /// <returns></returns>
+        public async Task UpdateDeploymentManifestFile(string saveCdkDirectoryFullPath)
+        {
+            if (string.IsNullOrEmpty(TargetApplicationFullPath))
+                throw new InvalidOperationException($"{nameof(TargetApplicationFullPath)} is null as part of the " +
+                    $"{nameof(DeploymentManifestEngine)} object");
+
+            var deploymentManifestFilePath = GetDeploymentManifestFilePath();
+            var saveCdkDirectoryRelativePath = _directoryManager.GetRelativePath(TargetApplicationFullPath, saveCdkDirectoryFullPath);
+
+            DeploymentManifestModel deploymentManifestModel;
+
+            if (_fileManager.Exists(deploymentManifestFilePath))
+            {
+                deploymentManifestModel = await ReadManifestFile(deploymentManifestFilePath);
+                deploymentManifestModel.DeploymentManifestEntries.Add(new DeploymentManifestEntry(saveCdkDirectoryRelativePath));
+            }
+            else
+            {
+                var deploymentManifestEntries = new List<DeploymentManifestEntry> { new DeploymentManifestEntry(saveCdkDirectoryRelativePath) };
+                deploymentManifestModel = new DeploymentManifestModel(deploymentManifestEntries);
+            }
+
+            var manifestFileJsonString = SerializeManifestModel(deploymentManifestModel);
+            await _fileManager.WriteAllTextAsync(deploymentManifestFilePath, manifestFileJsonString);
```

Comment:
There's a lot of file IO happening here. We should capture exceptions thrown here and throw back a meaningful expected exception.

Reply:
The `DeploymentManifestEngine` is just a wrapper class that supports the serialization and deserialization of the manifest file. Any exception thrown by this class should be caught inside the caller which calls the public methods of this class.

Reply:
The DeploymentManifestEngine has the sole responsibility of updating the manifest file. If it cannot do that job, it needs to throw a specific exception that can be caught by the caller. Either way, you are not catching any exceptions in the caller and any unexpected exception will bubble up to the user. I still think the DeploymentManifestEngine should be throwing an expected exception and that should be documented so that the caller knows that this method throws an exception. RIght now, the caller does not know whether the DeploymentManifestEngine throws exceptions. We need to separate responsibilities and we should not expect the caller to know more than it should.